src/discordtipbot/statistics.cr

```diff
@@ -0,0 +1,30 @@
+class Statistics
+  @db : DB::Database
+  @ttl : Time::Span
+
+  getter transactions : Int64?
+  getter total : BigDecimal?
+  getter tips : BigDecimal?
+  getter soaks : BigDecimal?
+  getter rains : BigDecimal?
+  getter last = Time.utc_now
+
+  def initialize(@db)
+    @ttl = 1.minutes
+    update_statistics
+  end
+
+  private def update_statistics
```

Comment:
This could all be offloaded to the DB and fetched with a single query using materialized views, and `update` would simply call `REFRESH MATERIALIZED VIEW`.